docs/en/plugins/grpc.md

```diff
@@ -1,20 +1,299 @@
+
+Next, you will need to create a PHP class that implements the `Pinger` service defined in the `.proto` file. This class
+should implement the `GRPC/Pinger/PingerInterface`.
+
+Here's an example:
+
+```php
+use Spiral\RoadRunner\GRPC;
+use GRPC\Pinger\PingerInterface;
+use GRPC\Pinger\PingRequest;
+use GRPC\Pinger\PingResponse;
+
+final class Pinger implements PingerInterface
+{
+    public function __construct(
+        private readonly HttpClientInterface $httpClient
+    ) {
+    }
+    
+    public function ping(GRPC\ContextInterface $ctx, PingRequest $in): PingResponse
+    {
+        $statusCode = $this->httpClient->get($in->getUrl())->getStatusCode();
+    
+        return new PingResponse([
+            'status_code' => $statusCode
+        ]);
+    }
+}
+```
+
+### Usage
+
+To use the `Pinger` service, you can create a PHP worker that registers the service with the gRPC server.
+
+Here's an example of how to do this:
+
+```php grpc-worker.php
+use GRPC\Pinger\PingerInterface;
+use Spiral\RoadRunner\GRPC\Server;
+use Spiral\RoadRunner\Worker;
+
+require __DIR__ . '/vendor/autoload.php';
+
+$server = new Server(null, [
+    'debug' => false, // optional (default: false)
+]);
+
+$server->registerService(PingerInterface::class, new Pinger(new HttpClient()));
+
+$server->serve(Worker::create());
+```
+
+After creating the worker, you need to configure RoadRunner to register the `proto/pinger.proto` service.
+
+Here's an example configuration:
+
+:::: tabs
+
+::: tab Server command
+
+You can define command to start server in the `server.command` section:. It will be used to start PHP workers for all
+registered plugins, such as `grpc`, `http`, `jobs`, etc.
+
+```yaml .rr.yaml
+version: "3"
+
+server:
+  command: "php grpc-worker.php"
+
+grpc:
+  listen: "tcp://127.0.0.1:9001"
+
+  proto:
+    - "proto/pinger.proto"
+```
+
+:::
+
+::: tab Worker commands
+You can also define command to start server in the `grpc.pool.command` section to separate server and grpc workers.
+
+```yaml .rr.yaml
+version: "3"
+
+server:
+  command: "php worker.php"
+
+grpc:
+  listen: "tcp://127.0.0.1:9001"
+
+  pool:
+    command: "php grpc-worker.php"
+
+  proto:
+    - "proto/pinger.proto"
+```
+
+:::
+::::
+
+> **Note**
+> You can define multiple proto files in the `proto` section.
+
 
-## Compiling proto sample:
+After configuring the server, you can start it using the following command:
 
+```terminal
+./rr serve
 ```
-protoc --plugin=protoc-gen-php-grpc --php_out=<OUTPUT DIRECTORY> --php-grpc_out=<OUTPUT DIRECTORY> simple.proto
+
+This will start the gRPC server and make the `Pinger` service available for remote clients to call. You can use any gRPC
+client library in any language that supports gRPC to call the `ping` method.
+
+## Metrics
+
+- [link](../lab/metrics.md)
+
+Example of grafana dashboard is
+available [here](https://github.com/roadrunner-server/roadrunner/blob/master/dashboards/grpc_dashboard.json).
+
+### Minimal dependencies:
```

Comment:
I guess we can delete this section, I created it for the custom plugins, but it's not needed anymore.